### PR TITLE
@damassi => responsive submission flow

### PR DESCRIPTION
**Pull request by sweir27**
Looks like a big diff but I'll point out the important bits. This PR makes the submission flow responsive using a similar technique as in prediction. Instead of using media queries (since there is a lot of shuffling of components on some of the forms), we have an `AppContainer` that dispatches updates to the window size and sets an `isMobile` property on the state. We pass `isMobile` into bem selectors and use it to hide/show components.

Artist selector:

Describe work:

Upload photos:

Thank you:

I also added a server-side route for the `thank_you` page for completeness.

### CODE REVIEW

File: desktop/apps/consignments2/components/submission_flow/index.js [hunk 1]
```diff
@@ -28,28 +34,37 @@ const mapStateToProps = (state) => {
   const {
     submissionFlow: {
       currentStep,
+      isMobile,
       steps
     }
   } = state
 
+  const describeWorkComponent = isMobile ? DescribeWorkMobile : DescribeWorkDesktop
```
Comment (sweir27) [inline desktop/apps/consignments2/components/submission_flow/index.js]: I decided to break out the `Mobile` and `Desktop` versions of this form after a bit of futzing-- it's basically the difference between having a bunch of logic inside one `DescribeWork` form, and have two separate components. For the mobile version, we want to require a different set of fields (and also hide a bunch of them). The inline logic was getting pretty messy, and there actually isn't that much logic _within_ the form, except for the `locationAutocomplete` which I extracted into its own component.

File: desktop/apps/consignments2/components/select_input/index.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ function SelectInput (props) {
                   href='#'
                   className='bordered-pulldown-active'
                   key={option}
-                  onClick={() => onChange(option)}
+                  onClick={(e) => onChange(option)}
```
Comment (damassi) [inline desktop/apps/consignments2/components/select_input/index.js]: Do you need the `e`vent here? Or just for correctness?

Comment (sweir27) [inline desktop/apps/consignments2/components/select_input/index.js]: Nope! Good catch. A remnant of me trying to debug something.

File: desktop/apps/consignments2/components/submission_flow/index.js (hunk 1 above)
Comment (damassi) [inline desktop/apps/consignments2/components/submission_flow/index.js]: I think the approach you took is clearer. No need to cram a bunch of responsive code into a single component when you can separate.

**Comment by sweir27** [comment]
Finally 🍏 !

**Pull request merged**